CMakeLists.txt

```diff
@@ -545,6 +545,14 @@ if(HPX_WITH_CUDA AND HPX_WITH_HIP)
   )
 endif()
 
+# ##############################################################################
+# HPX SYCL configuration
+# ##############################################################################
+set(SYCL_OPTION_STRING "Enable support for Sycl (default: OFF)")
+hpx_option(HPX_WITH_SYCL BOOL "${SYCL_OPTION_STRING}" OFF ADVANCED)
+set(HIPSYCL_OPTION_STRING "Use hipsycl cmake integration (default: OFF)")
+hpx_option(HPX_WITH_HIPSYCL BOOL "${HIPSYCL_OPTION_STRING}" OFF ADVANCED)
```

Comment:
Enabling `HIPSYCL` without enabling `SYCL` should produce a warning (or an error). Alternatively, it should enable SYCL (perhaps with a warning also).

Reply:
This is done in [HPX_SetupSYCL.cmake](https://github.com/G-071/hpx/blob/b4f1484d253fd514b66edc1a90441ca3537839ae/cmake/HPX_SetupSYCL.cmake#L19)